### PR TITLE
Feature/pam oauth2

### PR DESCRIPTION
**Pull request by CyberDem0n**

### CODE REVIEW

File: postgres-appliance/configure_spilo.py [hunk 1]
```diff
@@ -465,6 +465,25 @@ def write_ldap_configuration(placeholders, overwrite):
...
+    pam_oauth2_args = placeholders.get('PAM_OAUTH2') or ''
+    t = pam_oauth2_args.split()
+    if len(t) < 2:
+        logging.info("No PAM_OAUTH2 configuration was specified, skipping")
+        return
+
+    r = urlparse(t[0])
+    if not r.scheme or r.scheme != 'https':
+        logging.error('First argument of PAM_OAUTH2 must be a valid https url')
+        logging.info(r)
```
Comment (a1exsh) [inline postgres-appliance/configure_spilo.py]: Why not part of `logging.error`?

File: postgres-appliance/configure_spilo.py [hunk 2]
```diff
@@ -426,14 +426,11 @@ def write_etcd_configuration(placeholders, overwrite=False):
 def write_ldap_configuration(placeholders, overwrite):
     ldap_url = placeholders.get('LDAP_URL')
     if ldap_url is None:
-        logging.info("No LDAP_URL was specified, skipping LDAP configuration")
-        return
+        return logging.info("No LDAP_URL was specified, skipping LDAP configuration")
```
Comment (a1exsh) [inline postgres-appliance/configure_spilo.py]: That's a bit too smart ass, if you want my opinion. ;-)

Comment (a1exsh) [inline postgres-appliance/configure_spilo.py]: Also, my comment was about new code in `write_pam_oauth2_configuration()`...

Comment (CyberDem0n) [inline postgres-appliance/configure_spilo.py]: I know, fixed

**Comment by a1exsh** [comment]
:+1:

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**